### PR TITLE
[onnxifi] Add support for the newly added onnxGetEventState onnxifi method.

### PR DESCRIPTION
**Pull request by rdzhabarov**
cc: @yinghai @Maratyszcza

### CODE REVIEW

File: lib/Onnxifi/onnxifiGlow.cpp [hunk 1]
```diff
@@ -245,6 +245,23 @@ onnxWaitEvent(onnxEvent event) {
   return ONNXIFI_STATUS_SUCCESS;
 }
 
+/// Query ONNXIFI event state without blocking.
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
+  onnxGetEventState(
+    onnxEvent event,
+    onnxEventState* state) {
+  auto *glowEvent = reinterpret_cast<glow::onnxifi::EventPtr>(event);
```
Comment (Maratyszcza) [inline lib/Onnxifi/onnxifiGlow.cpp]: `static_cast` may be a better option. `onnxEvent` is just `void*`

File: lib/Onnxifi/onnxifiGlow.cpp [hunk 2]
```diff
@@ -245,6 +245,23 @@ onnxWaitEvent(onnxEvent event) {
...
+  if (!glowEvent) {
+    return ONNXIFI_STATUS_INVALID_EVENT;
+  }
+  if (!state) {
```
Comment (Maratyszcza) [inline lib/Onnxifi/onnxifiGlow.cpp]: This function must set `*state` to `ONNXIFI_EVENT_STATE_INVALID` if it fails. I suggest that you do the check `state != nullptr`, and immediately set `*state` to `ONNXIFI_EVENT_STATE_INVALID` if the check succeeds. You will later overwrite `*state` with the proper value before returning.

File: lib/Onnxifi/onnxifiGlow.cpp (hunk 1 above)
Comment (rdzhabarov) [reply lib/Onnxifi/onnxifiGlow.cpp]: @Maratyszcza Seems like the majority (all) onnxifi definitions for (event, graph, backend, etc) are void*. Is it safe to assume that it's not going to change? (There are other places in this implementation where reinterpret_cast could be replaced by static_cast, I'll change that in a separate commit).

File: lib/Onnxifi/onnxifiGlow.cpp (hunk 2 above)
Comment (rdzhabarov) [reply lib/Onnxifi/onnxifiGlow.cpp]: @Maratyszcza good point!

**Pull request merged**